lldb/include/lldb/Interpreter/Interfaces/ScriptedPlatformInterface.h

```diff
@@ -26,6 +26,10 @@ class ScriptedPlatformInterface : virtual public ScriptedInterface {
     return {llvm::make_error<UnimplementedError>()};
   }
 
+  llvm::SmallVector<llvm::StringLiteral> GetAbstractMethods() const override {
+    return {};
+  }
```

Comment:
If we don't implement this, we can't instantiate `ScriptedPlatformInterface` and that would be a problem for `ScriptedInterpreter::CreateScripted{Platform,Process,Thread}Interface`.

What do you think we should if we don't link against python and hence don't have a `ScriptedInterpreterPython` instance, to get each of the python interface instances ?

Reply:
This is probably a gap in my understanding, why do we instantiate `ScriptedPlatformInterface` instead of something like `ScriptedPythonPlatform`? The `Interface` at the end of `ScriptedPlatformInterface` to me feels like we shouldn't be creating objects of that type at all.

Reply:
I think I asked something similar in a previous PR (and forgot the answer). Give that this has come up repeatedly, it would be good to document this somewhere (for example in the relevant classes).

Reply:
The same way we instantiate a base `ScriptInterpreter` instance if we don't link with python. Because we use dynamic dispatch to get the right scripting interface from the script interpreter, we cannot make any of the Scripted affordance class abstract because we won't be able to instantiate them and hence, won't be able to implement `ScriptInterpreter::CreateScripted{Platform,Process,Thread}Interface`.

Another approach, that I'm not particularly keen on doing, is making them abstract and assert in the `ScriptInterpreter::CreateScripted{Platform,Process,Thread}Interface` methods. @JDevlieghere any thoughts ?

Reply:
Ok, I'll make the the base class abstract, and return a nullptr in `ScriptInterpreter::CreateScripted*Interface` in a follow-up, that should - hopefully - address all the confusion.

Reply:
@bulbazord @JDevlieghere Here! https://github.com/llvm/llvm-project/pull/71465